### ISSUE TITLE
No validation checks when overfit_pct is set

### ISSUE THREAD

**Issue by qmeeus** [PyTorchLightning/pytorch-lightning#1155]
## 🐛 Bug

When setting the `overfit_pct` to any value between 0 and 1 (exclusive) in trainer, the validation checks are disabled.

### To Reproduce

I have worked on a minimal example to reproduce the bug:

```python3
import pytorch_lightning as pl
import torch

class Dataset(torch.utils.data.Dataset):

    def __init__(self, input_dim, output_dim):
        super(Dataset, self).__init__()
        self.input_dim = input_dim
        self.output_dim = output_dim

    def __getitem__(self, idx):
        X = torch.rand(1, self.input_dim)
        y = torch.randint(0, self.output_dim, (1,))
        return X, y

    def __len__(self):
        return 1000

class Model(pl.LightningModule):

    def __init__(self, input_dim, output_dim):
        super(Model, self).__init__()
        self.layer = torch.nn.Linear(input_dim, output_dim)
        self.dataset = Dataset(input_dim, output_dim)

    def forward(self, x, y):
        yhat = torch.softmax(self.layer(x), -1)
        return F.nll_loss(logits, y)

    def train_dataloader(self):
        return torch.utils.data.DataLoader(self.dataset, batch_size=64)

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=1e-3)

    def training_step(self, batch, batch_idx):
        loss = self.forward(*batch)
        return {'loss': loss, 'log': {'loss': loss}}

    def validation_step(self, batch, batch_idx):
        loss = self.forward(*batch)
        return {'val_loss': loss, 'log': {'val_loss': loss}}


if __name__ == '__main__':
    model = Model(100, 10)
    trainer = pl.Trainer(overfit_pct=.01)
    trainer.fit(model)
```

### Expected behavior

Validation checks occur normally

### Environment
```bash
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Manjaro Linux
GCC version: (GCC) 8.3.0
CMake version: Could not collect

Python version: 3.7
Is CUDA available: No
CUDA runtime version: 10.2.89
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: /usr/lib/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] pytorch-lightning==0.7.1
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] mkl                       2020.0                      166  
[conda] pytorch                   1.4.0           py3.7_cuda10.1.243_cudnn7.6.3_0    pytorch
[conda] pytorch-lightning         0.7.1                    pypi_0    pypi
[conda] torchvision               0.5.0                py37_cu101    pytorch
```

**Comment by Borda** [comment]
@jeffling @awaelchli mind check?

**Comment by awaelchli** [comment]
Yes there is a bug here, but I had to fix @qmeeus's code sample to make it visible. 
The sanity validation checks run, but the validation at the end of the epoch doesn't.
Here is the fixed minimal code sample:
```
import pytorch_lightning as pl
import torch
import torch.nn.functional as F


class Dataset(torch.utils.data.Dataset):

    def __init__(self, input_dim, output_dim):
        super(Dataset, self).__init__()
        self.input_dim = input_dim
        self.output_dim = output_dim

    def __getitem__(self, idx):
        X = torch.rand(self.input_dim)
        y = torch.randint(0, self.output_dim, (1,))
        return X, y

    def __len__(self):
        return 1000


class Model(pl.LightningModule):

    def __init__(self, input_dim, output_dim):
        super(Model, self).__init__()
        self.layer = torch.nn.Linear(input_dim, output_dim)
        self.dataset = Dataset(input_dim, output_dim)

    def forward(self, x, y):
        logits = torch.softmax(self.layer(x), -1)
        return F.nll_loss(logits, y.flatten(0))

    def train_dataloader(self):
        return torch.utils.data.DataLoader(self.dataset, batch_size=64)

    def val_dataloader(self):
        return torch.utils.data.DataLoader(self.dataset, batch_size=64)

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=1e-3)

    def training_step(self, batch, batch_idx):
        loss = self.forward(*batch)
        return {'loss': loss, 'log': {'loss': loss}}

    def validation_step(self, batch, batch_idx):
        loss = self.forward(*batch)
        print('see that validation runs only in sanity check')
        return {'val_loss': loss, 'log': {'val_loss': loss}}

    def validation_end(self, outputs):
        loss = torch.stack([output['val_loss'] for output in outputs]).mean()
        return {'val_loss': loss, 'log': {'val_loss': loss}}


if __name__ == '__main__':
    model = Model(100, 10)
    trainer = pl.Trainer(overfit_pct=0.1, max_epochs=10)
    trainer.fit(model)
```
For the record, @qmeeus your code had these issues:
- No val_dataloader defined
- Wrong shapes returned in dataloader
- Wrong shape for nll_loss labels

**Comment by awaelchli** [comment]
Actually `overfit_pct `argument is not documented in the Trainer class. We should fix that and say that setting `overfit_pct `is the same as setting `train_percent_check`, `val_percent_check `and `test_percent_check`.

**Comment by awaelchli** [comment]
**False alarm!** Turns out it is simply because you chose a too small value for `overfit_pct`. 
Your dataset has size 1000, and dataloader has batch_size 64. 
1000 / 64 ~= 15 batches
When you choose overfit_pct = .01, then that gives 15 * 0.01 < 1 batch. 

@qmeeus Please let me know if it isn't clear. I think the behaviour of `overfit_pct `is correct.

**Comment by awaelchli** [comment]
@williamFalcon Should we make it so that `overfit_pct `does not round to 0 batches?

**Comment by qmeeus** [comment]
> **False alarm!** Turns out it is simply because you chose a too small value for `overfit_pct`.
> Your dataset has size 1000, and dataloader has batch_size 64.
> 1000 / 64 ~= 15 batches
> When you choose overfit_pct = .01, then that gives 15 * 0.01 < 1 batch.
> 
> @qmeeus Please let me know if it isn't clear. I think the behaviour of `overfit_pct `is correct.

Awesome, thanks !
